### PR TITLE
Issue #3219417 by SV: Add translation support for Topic content type

### PR DESCRIPTION
**Pull request by volodymyr-sydor-lemberg**
## Problem
As a CM+ I want to be able to add Topics in multiple languages

## Solution
Enable translation for Topic content type and all relevant fields

## Issue tracker
- https://www.drupal.org/project/social/issues/3219417
- https://getopensocial.atlassian.net/browse/YANG-5701

## How to test
- [ ] Using the latest version of Open Social with the social_topic module enabled
- [ ]  Login as an admin
- [ ]  Make sure you have at least two languages installed
- [ ]  Add an "Topic" and attach it to any group
- [ ]  Add translation for the created topic
- [ ]  Visit pages with the topic

## Release notes
 Add translation compatibility for Topic node type.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

File: modules/social_features/social_topic/config/update/social_topic_update_8910.yml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                table: node_field_data
+                value: '1'
+        group_topics_block:
+          display_options:
+            rendering_language: '***LANGUAGE_language_interface***'
+        page_latest_topics:
+          display_options:
+            rendering_language: '***LANGUAGE_language_interface***'
+            filters:
+              default_langcode:
```
[the remark this answers is not in the document]
Comment (volodymyr-sydor-lemberg) [reply modules/social_features/social_topic/config/update/social_topic_update_8910.yml]: Looks like duplicated, removed it

File: modules/social_features/social_topic/config/update/social_topic_update_8910.yml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                value: '1'
+views.view.topics:
+  expected_config: {  }
+  update_actions:
+    add:
+      display:
+        block_user_topics:
+          display_options:
+            filters:
+              default_langcode:
```
[the remark this answers is not in the document]
Comment (volodymyr-sydor-lemberg) [reply modules/social_features/social_topic/config/update/social_topic_update_8910.yml]: removed it too

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Review by ribel — CHANGES_REQUESTED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

File: modules/social_features/social_topic/social_topic.install [hunk 3]
```diff
@@ -412,3 +412,17 @@ function social_topic_update_8909() {
     }
   }
 }
+
+/**
+ * Add translation compatibility for Topic fields and views.
+ */
+function social_topic_update_8910() {
```
[the remark this answers is not in the document]
Comment (volodymyr-sydor-lemberg) [reply modules/social_features/social_topic/social_topic.install]: done

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2387-cucch6mitst2f7bwnpjn9fgxr8p5exax.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60cc6d43c17bbe5a057c40d6

**Review by ribel — APPROVED** [review]

**Pull request merged**